www/src/components/docs-markdown-page.js

```diff
@@ -0,0 +1,141 @@
+/** @jsx jsx */
+import { jsx } from "theme-ui"
+import React from "react"
+import { MDXRenderer } from "gatsby-plugin-mdx"
+import { mediaQueries } from "gatsby-design-tokens/dist/theme-gatsbyjs-org"
+
+import PageWithSidebar from "../components/page-with-sidebar"
+import PageMetadata from "../components/page-metadata"
+import DocSearchContent from "../components/docsearch-content"
+import TableOfContents from "../components/docs-table-of-contents"
+import FooterLinks from "../components/shared/footer-links"
+import Breadcrumb from "../components/docs-breadcrumb"
+import Container from "../components/container"
+import MarkdownPageFooter from "../components/markdown-page-footer"
+import PrevAndNext from "../components/prev-and-next"
+
+const containerStyles = {
+  // we need to account for <Container>'s horizontal padding of
+  // `space[6]` each (1.5rem), plus add a fluffy `space[9]`
+  // of whitespace in between main content and TOC
+  //
+  // could be much cleaner/clearer, please feel free to improve 🙏
+  maxWidth: t =>
+    `calc(${t.sizes.mainContentWidth.withSidebar} + ${t.sizes.tocWidth} + ${t.space[9]} + ${t.space[9]} + ${t.space[9]})`,
+  px: 9,
+}
+
+function DocsMarkdownPage({
+  page,
+  location,
+  prev,
+  next,
+  items: itemsProp = [],
+  depth: depthProp,
+  children,
+  ...others
+}) {
+  const getTOCItems = () => {
+    if (page.frontmatter.disableTableOfContents) {
+      return []
+    }
+
+    const items = page.tableOfContents.items || [];
+
+    return [...items, ...itemsProp]
```

Comment:
I would put this logic in `template-api-markdown` instead. Then you can put the defaults right in the destructure:

```
function DocsMarkdownPage({
  page,
  tableOfContentsItems = page.tableOfContents.items,
  tableOfContentsDepth = page.frontmatter.tableOfContentsDepth,
})